subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -79,6 +84,13 @@
 
     private FileCollection annotationProcessorPath;
 
+    private final PropertyState<File> annotationProcessorGeneratedSourcesDirectory;
```

Comment:
Would it make sense to add another nested object of type `AnnotationProcessorOptions` which would have the `path` and the `generatedSourcesDirectory` as properties? We could even move this object to `JavaCompile` I guess.

Reply:
Let's do something like this again when we have a plan to rework compiler options.